modules/@apostrophecms/piece-type/ui/apos/components/AposImageCropper.vue

```diff
@@ -0,0 +1,107 @@
+<template>
+  <div class="apos-image-cropper">
+    <cropper
+      ref="cropper"
+      :src="imgInfos.url"
+      @change="onChange"
+      :default-size="defaultSize"
+    />
+  </div>
+</template>
+
+<script>
+import { debounce } from 'Modules/@apostrophecms/ui/utils';
+import { Cropper } from 'vue-advanced-cropper';
+import 'vue-advanced-cropper/dist/style.css';
+
+export default {
+  components: {
+    Cropper
+  },
+  props: {
+    imgInfos: {
+      type: Object,
+      required: true
+    },
+    docFields: {
+      type: Object,
+      required: true
+    }
+  },
+  emits: [ 'change' ],
+  data: () => ({
+    isUpdatingCoordinates: false
+  }),
+  watch: {
+    docFields: {
+      deep: true,
+      handler(newVal, oldVal) {
+        if (this.checkCoordinatesDiff(newVal.data, oldVal.data)) {
+          const {
+            width, height, left, top
+          } = newVal.data;
+
+          this.isUpdatingCoordinates = true;
+          this.setCoordinates({
+            width,
+            height,
+            left,
+            top
+          });
+        }
+      }
+    }
+  },
+  created () {
+    this.setCoordinates = debounce(
+      (coordinates) => {
+        this.$refs.cropper.setCoordinates(coordinates);
+      }, 500
+    );
+  },
+  methods: {
+    defaultSize({ imageSize, visibleArea }) {
+      return {
+        width: (visibleArea || imageSize).width,
+        height: (visibleArea || imageSize).height
```

Comment:
I think this part should be modified, but wanna be sure with you @boutell. We want to apply a cropping that would exist on the attachment document itself right? If no cropping we take the initial size of the image as we do here.